modules/gcloud/gcloud.go

```diff
@@ -72,6 +73,18 @@ func WithProjectID(projectID string) Option {
 	}
 }
 
+// WithDataYamlFile seeds the Bigquery project for the GCloud container
+// with the local path to the data yaml file, which is used to copy the file to the container,
+// and then process the file to seed the Bigquery project.
+//
+// Other GCloud containers will ignore this option.
+// If this option is passed multiple times, the last file will be used.
+func WithDataYamlFile(f string) Option {
```

Comment:
question: use a `ContainerCustomiser` and pass a `io.Reader` instead?

Reply:
If we pass a customiser here, we'd need to append the file to the req.Files field. If users pass the option multiple times, as advised in the docs, the last one will win, but the code would unnecessarily copy multiple files to the container. That's why I went with the Option, to store the file (the reader) into the options type and then process it when needed.

Reply:
Yes but that too is confusing, I would suggest that two uses should error to make it clear to the user what they did wrong.

```go
func WithDataYAML(r io.Reader) testcontainers.CustomizeRequestOption {
	return func(req *testcontainers.GenericContainerRequest) error {
		const dataFlag = "--data-from-yaml"
		for i, flag := range req.Cmd {
			if flag == dataFlag {
				if i+1 < len(req.Cmd) {
					return fmt.Errorf("%s already set to %q", dataFlag, req.Cmd[i+1])
				}
				return fmt.Errorf("%s already set without a value", dataFlag)
			}
		}

		containerPath := "/testcontainers-data.yaml"
		req.Cmd = append(req.Cmd, dataFlag, containerPath)

		req.Files = append(req.Files, testcontainers.ContainerFile{
			Reader:            r,
			ContainerFilePath: containerPath,
			FileMode:          0o644,
		})

		return nil
	}
}
```

Updated tests to match using embed to simplify:
```go

//go:embed testdata/data.yaml
var dataYaml []byte

func TestBigQueryWithDataYamlFile(t *testing.T) {
	ctx := context.Background()

	t.Run("valid", func(t *testing.T) {
		bigQueryContainer, err := gcloud.RunBigQuery(
			ctx,
			"ghcr.io/goccy/bigquery-emulator:0.6.1",
			gcloud.WithProjectID("test"),
			gcloud.WithDataYAML(bytes.NewReader(dataYaml)),
		)
		testcontainers.CleanupContainer(t, bigQueryContainer)
		require.NoError(t, err)

		projectID := bigQueryContainer.Settings.ProjectID

		opts := []option.ClientOption{
			option.WithEndpoint(bigQueryContainer.URI),
			option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())),
			option.WithoutAuthentication(),
			internaloption.SkipDialSettingsValidation(),
		}

		client, err := bigquery.NewClient(ctx, projectID, opts...)
		require.NoError(t, err)
		defer client.Close()

		selectQuery := client.Query("SELECT * FROM dataset1.table_a where name = @name")
		selectQuery.QueryConfig.Parameters = []bigquery.QueryParameter{
			{Name: "name", Value: "bob"},
		}
		it, err := selectQuery.Read(ctx)
		require.NoError(t, err)

		var val []bigquery.Value
		for {
			err := it.Next(&val)
			if errors.Is(err, iterator.Done) {
				break
			}
			require.NoError(t, err)
		}

		require.Equal(t, int64(30), val[0])
	})

	t.Run("multi-value-set", func(t *testing.T) {
		bigQueryContainer, err := gcloud.RunBigQuery(
			ctx,
			"ghcr.io/goccy/bigquery-emulator:0.6.1",
			gcloud.WithProjectID("test"),
			gcloud.WithDataYAML(bytes.NewReader(dataYaml)),
			gcloud.WithDataYAML(bytes.NewReader(dataYaml)),
		)
		testcontainers.CleanupContainer(t, bigQueryContainer)
		require.EqualError(t, err, `--data-from-yaml already set to "/testcontainers-data.yaml"`)
	})

	t.Run("multi-value-not-set", func(t *testing.T) {
		noValueOption := func() testcontainers.CustomizeRequestOption {
			return func(req *testcontainers.GenericContainerRequest) error {
				req.Cmd = append(req.Cmd, "--data-from-yaml")
				return nil
			}
		}

		bigQueryContainer, err := gcloud.RunBigQuery(
			ctx,
			"ghcr.io/goccy/bigquery-emulator:0.6.1",
			noValueOption(),
			gcloud.WithProjectID("test"),
			gcloud.WithDataYAML(bytes.NewReader(dataYaml)),
		)
		testcontainers.CleanupContainer(t, bigQueryContainer)
		require.EqualError(t, err, `--data-from-yaml already set without a value`)
	})
}
```

Reply:
Cool, I like the error if the option is passed more than once. Will implement the embed thing (always forget it's available!)

Reply:
One thing: at the moment we convert this option into a Customiser, then any other GCloud container will be able to use it. With the options approach, it's only processed in the right place, but I'm realising it's not following the "return error" API for the options. Should we add a BC here?

At some point we should tackle an specialisation of the current GCloud containers, or at least the options, so that they can be only used by their respective owners. Using packages for each container could be an option, so users will know the option belongs to a given package in an explicit manner.

Reply:
It's already documented that its only valid for bigquery. We could add a check based on an env var if you wanted to make it explicit, returning an error if its used in an unexpected way, however I would expect you get an error when used incorrectly anyway due to basic arg validation from the containers themselves.

Previously the arg would be silently ignored by other containers, which leads the user to believe its been applied making it confusing. I always prefer things to error if I use them wrong as it doesn't leave you scratching your head wondering why something didn't work.

Reply:
That's fair. In theory, any module is exporting options, so why not breaking my mysql container with a neo4j option? 🤷 

So failing is fine then, thanks!